evm/src/cpu/kernel/asm/curve/bn254/field_arithmetic/frobenius.asm

```diff
@@ -0,0 +1,260 @@
+global test_frob_fp12_1:
+    // stack: ptr, f, ptr
+    %store_fp12
+    // stack:         ptr
+    %frob_fp12_1
+    // stack:         ptr
+    %load_fp12
+    %jump(0xdeadbeef)
+
+global test_frob_fp12_2:
+    // stack: ptr, f, ptr
+    %store_fp12
+    // stack:         ptr 
+    DUP1
+    // stack:    ptr, ptr
+    %frob_fp12_2_
+    // stack:         ptr
+    %load_fp12
+    %jump(0xdeadbeef)
+
+global test_frob_fp12_3:
+    // stack: ptr, f, ptr
+    %store_fp12
+    // stack:         ptr
+    %frob_fp12_3
+    // stack:         ptr
+    %load_fp12
+    %jump(0xdeadbeef)
+
+global test_frob_fp12_6:
+    // stack: ptr, f, ptr
+    %store_fp12
+    // stack:         ptr
+    %frob_fp12_6
+    // stack:         ptr
+    %load_fp12
+    %jump(0xdeadbeef)
```

Comment:
You can do this from within Rust, so you can take it out of the kernel codebase.